gitlab/tests/test_config.py

```diff
@@ -192,6 +196,17 @@ def test_valid_data(m_open, path_exists):
     assert 2 == cp.timeout
     assert True == cp.ssl_verify
 
+    fd = io.StringIO(valid_config)
+    fd.close = mock.Mock(return_value=None)
+    m_open.return_value = fd
+    cp = config.GitlabConfigParser(gitlab_id="five")
+    assert "five" == cp.gitlab_id
+    assert "https://five.url" == cp.url
+    assert None == cp.private_token
+    assert "foobar" == cp.oauth_token
+    assert 2 == cp.timeout
+    assert True == cp.ssl_verify
+
```

Comment:
Thanks for adding tests for this :)

This test is really growing now, it might make sense to split it into a new `test_data_from_helper`. Then you could check: what error the user will see if the subprocess exits with non-zero, if it returns nothing, in case there's nothing given in the config after the prefix, etc.

Most of the other stuff in this class is wrapped in blanket try/except blocks so I think if we don't do it for helpers, it should be tested well.

Reply:
Improved the test and adapted to changed helper.